src/shim/runtime/v2/rune/agent/client/client.go

```diff
@@ -0,0 +1,166 @@
+// The codebase is inherited from kata-containers with the modifications.
+
+package client
+
+import (
+	"context"
+	"net"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc/codes"
+	grpcStatus "google.golang.org/grpc/status"
+
+	agentgrpc "github.com/confidential-containers/enclave-cc/src/shim/runtime/v2/rune/agent/grpc"
+	"github.com/containerd/ttrpc"
+)
+
+var defaultDialTimeout = 30 * time.Second
+
+var agentClientFields = logrus.Fields{
+	"name":   "agent-client",
+	"pid":    os.Getpid(),
+	"source": "agent-client",
+}
+
+const (
+	TcpSocketScheme = "tcp"
+)
+
+var agentClientLog = logrus.WithFields(agentClientFields)
+
+// AgentClient is an agent gRPC client connection wrapper for agentgrpc.AgentServiceClient
+type AgentClient struct {
+	ImageServiceClient agentgrpc.ImageService
+	conn               *ttrpc.Client
+}
+
+// NewAgentClient creates a new agent gRPC client and handles tcp address.
+//
+// Supported sock address format is:
+//   - tcp://<ip>:<port>
+func NewAgentClient(ctx context.Context, sock string, timeout uint32) (*AgentClient, error) {
+	grpcAddr, parsedAddr, err := parse(sock)
```

Comment:
Is there a need for `parse()` anymore? It looks it just changes `tcp://foobar` to `tcp:foobar` from which we strip `tcp:` prefix. we could just call `TcpDialer()` (but maybe rename it)  directly instead?

Reply:
Hi, Mikko. In order to support both TCP and UNIX socket, it is better to add parse().  Libos may not support unix socket in some situations, user can change to TCP socket quickly.

Reply:
this initial version comes with TCP only so I was proposing to simplify code for that.